### PR TITLE
CB-16762 Add schedule and unschedule endpoint to consumption service

### PR DESCRIPTION
**Pull request by nadamdb**
The schedule and unschedule endpoints are used to schedule and unschedule quartz jobs in the cloud consumption service that periodically collect cloud storage consumption for a given environment and storage location.

See detailed description in the commit message.

### CODE REVIEW

File: cloud-consumption-api/src/main/java/com/sequenceiq/consumption/api/v1/consumption/endpoint/ConsumptionEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.sequenceiq.consumption.api.v1.consumption.model.request.StorageConsumptionRequest;
+import com.sequenceiq.consumption.api.doc.ConsumptionOpDescription;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+
+@Path("/v1/consumption")
+@RetryAndMetrics
+@Consumes(MediaType.APPLICATION_JSON)
+@Api(value = "/consumption", protocols = "http,https", consumes = MediaType.APPLICATION_JSON)
```
Comment (bergerdenes) [inline cloud-consumption-api/src/main/java/com/sequenceiq/consumption/api/v1/consumption/endpoint/ConsumptionEndpoint.java:21]: different path

File: cloud-consumption-api/src/main/java/com/sequenceiq/consumption/api/v1/consumption/endpoint/ConsumptionEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+public interface ConsumptionEndpoint {
+
+    @POST
+    @Path("schedule/storage")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = ConsumptionOpDescription.SCHEDULE_STORAGE, produces = MediaType.APPLICATION_JSON, nickname = "scheduleStorageCollection")
+    void scheduleStorageUsageCollection(@Valid StorageConsumptionRequest request);
```
Comment (bergerdenes) [inline cloud-consumption-api/src/main/java/com/sequenceiq/consumption/api/v1/consumption/endpoint/ConsumptionEndpoint.java:28]: @NotNull annotation might be beneficial as well

File: cloud-consumption-api/src/main/java/com/sequenceiq/consumption/api/v1/consumption/endpoint/ConsumptionEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    @DELETE
+    @Path("unschedule/storage")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = ConsumptionOpDescription.UNSCHEDULE_STORAGE, produces = MediaType.APPLICATION_JSON, nickname = "unscheduleStorageCollection")
+    void unscheduleStorageUsageCollection(@Valid StorageConsumptionRequest request);
```
Comment (bergerdenes) [inline cloud-consumption-api/src/main/java/com/sequenceiq/consumption/api/v1/consumption/endpoint/ConsumptionEndpoint.java:34]: @NotNull

File: cloud-consumption-api/src/main/java/com/sequenceiq/consumption/api/v1/consumption/model/request/StorageConsumptionRequest.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import com.sequenceiq.cloudbreak.auth.crn.CrnResourceDescriptor;
+import com.sequenceiq.cloudbreak.validation.ValidCrn;
+import com.sequenceiq.consumption.api.doc.ConsumptionModelDescription;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel(value = "StorageConsumptionRequest")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class StorageConsumptionRequest extends ConsumptionBaseRequest {
```
Comment (bergerdenes) [inline cloud-consumption-api/src/main/java/com/sequenceiq/consumption/api/v1/consumption/model/request/StorageConsumptionRequest.java:15]: Just a discussion-opening question: is it the same request class for both schedule and unschedule.
Also, the naming is not consistent with the endpoint method names where this class is used as parameter type:  StorageUsage vs StorageConsumption

**Review by bergerdenes — CHANGES_REQUESTED** [review]

File: cloud-consumption-api/src/main/java/com/sequenceiq/consumption/api/v1/consumption/model/request/StorageConsumptionRequest.java (hunk 4 above)
Comment (nadamdb) [reply cloud-consumption-api/src/main/java/com/sequenceiq/consumption/api/v1/consumption/model/request/StorageConsumptionRequest.java:15]: Good point, created different requests for the two actions and made the method/class naming consistent

**Review by bergerdenes — APPROVED** [review]

File: cloud-consumption/src/main/java/com/sequenceiq/consumption/endpoint/ConsumptionController.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import com.sequenceiq.consumption.api.v1.consumption.endpoint.ConsumptionEndpoint;
+import com.sequenceiq.consumption.api.v1.consumption.model.request.StorageConsumptionScheduleRequest;
+import com.sequenceiq.consumption.api.v1.consumption.model.request.StorageConsumptionUnscheduleRequest;
+import com.sequenceiq.consumption.domain.Consumption;
+import com.sequenceiq.consumption.service.ConsumptionService;
+
+@Controller
+@Transactional(Transactional.TxType.NEVER)
+@AccountEntityType(Consumption.class)
+public class ConsumptionController implements ConsumptionEndpoint {
```
Comment (horadla23) [inline cloud-consumption/src/main/java/com/sequenceiq/consumption/endpoint/ConsumptionController.java:21]: missing authorization.
also please add `EnforceAuthorizationAnnotationsTest` to the consumption service, other modules has it

**Review by horadla23 — CHANGES_REQUESTED** [review]

**Review by horadla23 — APPROVED** [review]

**Pull request merged**